Sources/OpenPass/IDTokenValidation.swift

```diff
@@ -0,0 +1,93 @@
+//
+//  IDTokenValidation.swift
+//
+// MIT License
+//
+// Copyright (c) 2024 The Trade Desk (https://www.thetradedesk.com/)
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+//
+
+import Foundation
+
+/// Validates an IDToken against JWKS.
+protocol IDTokenValidation {
+    func validate(_ token: IDToken, jwks: JWKS) throws -> Bool
+}
+
+/// Validates the following:
+/// - Token is not expired
+/// - Token issue date is not in the past
+/// - Token signature is correct according to JWKS
+///  https://openid.net/specs/openid-connect-core-1_0.html#IDTokenValidation
+internal struct IDTokenValidator: IDTokenValidation {
+    /// Set a specific leeway window in seconds in which the Expires At ("exp") Claim will still be valid.
+    private let expiresAtLeeway: Double = 0
```

Comment:
Not important, but just an observation... On Android, we actually expose both the `expiresAtLeeway` and `issuedAtLeeway` publicly on a per-flowclient basis. I'm trying to remember the reasoning, but I think it was just the fact that time synchronisation is tough 😅